### PR TITLE
Block RUSH_TEMP_FOLDER usage in workspace installs

### PR DESCRIPTION
**Pull request by D4N14L**
## Summary

Mark the `RUSH_TEMP_FOLDER` environment variable for deprecation and block usage of the variable in workspace installs.

## Details

It was found in #2566 that the RUSH_TEMP_FOLDER variable was being accepted when using workspace installs when workspace installs were never compatible with the variable in the first place. This PR marks the variable for deprecation in a future version of Rush, and prevents usage on workspace installs starting immediately.

## How it was tested

Set `RUSH_TEMP_FOLDER` in a workspace environment and confirmed that the install failed to begin.

### CODE REVIEW

**Comment by iclanton** [comment]
I think the original usage of the `RUSH_TEMP_FOLDER` still makes sense. We wanted a way to put the installed modules somewhere outside of the repo to easier to clean up and/or share on a machine.

Disabling moving the whole `common/temp` folder makes sense for workspace repos, but we should provide a way to override the store location, ideally both via an environment variable and with a setting in `~/.rush-user/settings.json`

**Comment by D4N14L** [comment]
> I think the original usage of the `RUSH_TEMP_FOLDER` still makes sense. We wanted a way to put the installed modules somewhere outside of the repo to easier to clean up and/or share on a machine.
> 
> Disabling moving the whole `common/temp` folder makes sense for workspace repos, but we should provide a way to override the store location, ideally both via an environment variable and with a setting in `~/.rush-user/settings.json`

Agreed. The setting to override the store location (`RUSH_PNPM_STORE_PATH`) is still present, this only removes `RUSH_TEMP_FOLDER`. I also chose to leave the env variable functional for non-workspace installs. I can remove the deprecation tag on the var and replace it with a note though, since it's not _actually_ deprecated.

File: apps/rush-lib/src/logic/installManager/WorkspaceInstallManager.ts [hunk 1]
```diff
@@ -66,6 +67,11 @@ export class WorkspaceInstallManager extends BaseInstallManager {
   ): Promise<{ shrinkwrapIsUpToDate: boolean; shrinkwrapWarnings: string[] }> {
     const stopwatch: Stopwatch = Stopwatch.start();
 
+    // Block use of the RUSH_TEMP_FOLDER environment variable
+    if (EnvironmentConfiguration.rushTempFolderOverride !== undefined) {
+      throw new Error('The RUSH_TEMP_FOLDER environment variable is not compatible with workspace installs.');
```
Comment (iclanton) [inline apps/rush-lib/src/logic/installManager/WorkspaceInstallManager.ts:72]: Can you have this suggest using the PNPM store env var?

**Comment by D4N14L** [comment]
> ideally both via an environment variable and with a setting in `~/.rush-user/settings.json`

I decided to leave this alone for now. @octogonz has been musing about moving the `pnpmOptions` (and other package manager specific options) into their own file(s?), and I feel like this would be something that we would want to go in there as well.

**Review by iclanton — APPROVED** [review]

**Comment by octogonz** [comment]
Thanks for fixing this!

**Pull request merged**